formcontent/formcontent.go

```diff
@@ -10,73 +10,87 @@ import (
 )
 
 type Form struct {
-	formFile string
+	multipartWriter *multipart.Writer
+	body            *os.File
 }
 
 type ContentSubmission struct {
 	Length      int64
-	Content     io.ReadCloser
+	Content     io.Reader
 	ContentType string
 }
 
-func NewForm(formFile string) Form {
-	return Form{
-		formFile: formFile,
+func NewForm() (Form, error) {
+	body, err := ioutil.TempFile("", "")
+	if err != nil {
+		return Form{}, err
 	}
+
+	return Form{
+		multipartWriter: multipart.NewWriter(body),
+		body:            body,
+	}, nil
 }
 
-func (f Form) Create(path string) (ContentSubmission, error) {
-	originalContent, err := os.Open(path)
+func (f Form) Create() (ContentSubmission, error) {
```

Comment:
Can we rename this to something like `Finalize`? Want to make it clear that one should only call this after they are done adding all of the content in the form.